### PR TITLE
fix(core): fix infinite retry when workspace not found

### PR DESCRIPTION
**Pull request by EYHN**

### CODE REVIEW

**Comment by EYHN** [comment]
* **#8679** <a href="https://app.graphite.dev/github/pr/toeverything/AFFiNE/8679?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `canary`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @EYHN and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: packages/frontend/core/src/desktop/pages/workspace/index.tsx [hunk 1]
```diff
@@ -84,13 +84,16 @@ export const Component = (): ReactElement => {
...
-      const timer = setInterval(
-        () => workspacesService.list.revalidate(),
-        5000
-      );
+      const timer = setInterval(() => {
+        if (retryTimes > 0) {
+          workspacesService.list.revalidate();
+          retryTimes--;
+        }
+      }, 5000);
```
[the remark this answers is not in the document]
Comment (pengx17) [reply packages/frontend/core/src/desktop/pages/workspace/index.tsx]: I think this is valid comment. revalidate will reset `listLoading` and cause the effect to rerun and reset the timer

[the remark this answers is not in the document]
Comment (EYHN) [reply packages/frontend/core/src/desktop/pages/workspace/index.tsx]: fixed

File: packages/frontend/core/src/desktop/pages/workspace/index.tsx [hunk 2]
```diff
@@ -84,13 +84,16 @@ export const Component = (): ReactElement => {
     }
   }, [listLoading, meta, workspacesService]);
 
-  // if workspace is not found, we should revalidate in interval
+  // if workspace is not found, we should retry
+  const retryTimesRef = useRef(3);
```
Comment (pengx17) [inline packages/frontend/core/src/desktop/pages/workspace/index.tsx:88]: do we need to reset the timer when workspace id changed?

**Review by pengx17 — APPROVED** [review]

**Pull request merged**